ngraph/test/backend/prelu.in.cpp

```diff
@@ -0,0 +1,349 @@
+// Co pyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
+
+NGRAPH_TEST(${BACKEND_NAME}, prelu)
+{
+    Shape shape{3, 2};
+    Shape rshape{3};
+    auto A = make_shared<op::Parameter>(element::f32, shape);
+    auto B = make_shared<op::Parameter>(element::f32, rshape);
+    auto prelu = make_shared<op::PRelu>(A, B);
+    auto f = make_shared<Function>(NodeVector{prelu}, ParameterVector{A, B});
+    std::vector<float> a{-2, 3, -2, 1, -1, 0};
+    std::vector<float> b{0, 0.5, 1};
+
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_multiple_inputs<float>({a, b});
+    test_case.add_expected_output<float>(vector<float>{0, 3, -1, 1, -1, 0});
+    EXPECT_THROW(test_case.run(), std::exception);
+}
```

Comment:
I am not sure that this test is correct. If I am not mistaken the CPU plugin applies the slope per channel. So the result should be [0,0, -1, 0.5, -1, 0].

@IvanNovoselov , @mitruska , am I right?

Reply:
@lazarevevgeny, this test is not clear to me, because A contains only one channel, but B has 3 values. Shouldn't the shape of A be {3,1,2}?
Assuming that the number of channels is 3, the original expected output looks good because positive values should not be multiplied by slopes: {0*(-2), 3, 0.5*(-2), 1, 1*(-1), 0}.

Reply:
Oh, yes. Sorry for confusion. This test is definitely invalid.
We should support cases like: 
```
input [3,2], slope [2]. (Per channel)
input [3,2,5], slope[2], (Per channel)
input [3,2,5,7], slope[2], (Per channel)
input [3,2,5], slope[3,2,5] (equal shapes)
input [3,2,5], slope[1],
```

Reply:
First of all I can see that this test `EXPECT_THROW` so it is expected to fail by definition 
(anyway this test is excluded in manifest for CPU, because with 2021.3 it throws:
`Eltwise node with name PRelu_9783 has invalid input/output dims configuration.`)
I agree it is confusing, so we should update at least the name of this test.

No sure if I'm up to date with the latest discussion but as long as we stay with CPU PRelu (2021.3) and apply numpy broadcasting, then shapes from this test [3, 2] and [3] are simply not broacastable (numpy broadcast also would throw).

From the [numpy broadcast](https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules) point of view, 1D is broadcastable with ND only if the last dimension of ND match the 1D or is equal to 1.

```
input [3,2], slope [2] or [3, 1] or [3, 2] or [1]
input [3,2,5], slope [5] or [2, 1] or [3, 1, 1] but also [3, 2, 1] or [3, 1, 5] or  ... [3, 2, 5] or [1]
input [3,2,5,7], slope[7] or [5, 1] or [2, 1, 1] ... or [1]
```
And the (2021.3) PRelu operation is performed element-wise with numpy broadcasted input shapes, I would say without care about layout or channel dim position.

Reply:
We have mtcnn-o model from caffe where the IR looks the following way:
```
        <layer id="6" name="prelu1" type="PReLU" version="opset1">
            <input>
                <port id="0">
                    <dim>1</dim>
                    <dim>32</dim>
                    <dim>46</dim>
                    <dim>46</dim>
                </port>
                <port id="1">
                    <dim>32</dim>
                </port>
            </input>
            <output>
                <port id="2" precision="FP32" names="conv1">
                    <dim>1</dim>
                    <dim>32</dim>
                    <dim>46</dim>
                    <dim>46</dim>
                </port>
            </output>
        </layer>
```

And this model is successfully converted and inferred. So this is why I added test cases when per-channel broadcasting is needed.

Reply:
Looks like:
https://github.com/openvinotoolkit/openvino/blob/c6c38d90ca54ec37aa0be08be15fc69201eb5e7c/ngraph/test/backend/prelu.in.cpp#L282-L286

Reply:
I see we are back with discussion about channels. Let settle once again do we want behavior from 2021.3 or broadcasting by channel. Broadcasting by channel is not supported in 2021.3 and that's why `prelu_1d_C_slope` is disabled in manifest and it expect to throw in INTERPRETER.
@lazarevevgeny , Is mtcnn-o model pass in master or in 2021.3 ?

Reply:
Please add same tests as `prelu_1d_C_slope` and `prelu_1d_W_slope` but with Constant Slope inptut:
```
    auto A = make_shared<op::Parameter>(element::f32, shape_a);
    auto SLOPE = make_shared<op::Constant>(element::f32, shape_slope);
    auto f = make_shared<Function>(make_shared<op::PRelu>(A, SLOPE), ParameterVector{A});
```